File: 205-isomorphic-strings/205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mapper = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if (s[i] not in mapper) and (t[i] not in mapper.values()):
                mapper[s[i]] = t[i]
        
        for i in range(len(s)):
            if (s[i] not in mapper):
                return False
            
            if (mapper[s[i]] != t[i]):
                return False
        return True

File: 205-isomorphic-strings/205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    const mapper = {};
    if (s.length !== t.length){
        return false;
    };
    
    for (let i=0; i<s.length; i++){
        if (!mapper.hasOwnProperty(s[i]) && !Object.values(mapper).includes(t[i])){
            mapper[s[i]] = t[i];
        };
    };
    
    for (let i=0; i<s.length; i++){
        if (mapper[s[i]] !== t[i]){
            return false;
        };
    };
    return true;
};